scripts/init-repo/functions.ps1

```diff
@@ -0,0 +1,64 @@
+####################################################
+# functions                                        #
+####################################################
+
+function PromptUserYN
+{
+    [cmdletbinding()]
+    param
+    (
+        [string]
+        $Message,
+
+        [string]
+        $Question
+    )
+    if ($script:PromptUserAlwaysYes)
+    {
+        return $true;
+    }
+
+    $choices  = '&Yes', '&No', 'Yes to &all';
+    $decision = $Host.UI.PromptForChoice($Message, $Question, $choices, 1);
+    if ($decision -eq 1)
+    {
+        return $false;
+    }
+    if ($decision -eq 2)
+    {
+        $script:PromptUserAlwaysYes = $true;
+    }
+    return $true;
+}
+
+function Install-Choco
+{
+    if ((Get-Command -Name choco -ErrorAction SilentlyContinue) -eq $null)
+    {
+        Set-ExecutionPolicy Bypass -Scope Process -Force;
+        iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'));
+        if ((Get-Command -Name choco -ErrorAction SilentlyContinue) -eq $null)
+        {
+            Write-Warning 'Unable to find [choco] command. Try restarting the powershell console and run again the whole script.'
+            return;
+        }
+    }
+}
+
+function ErrorOccurred
```

Comment:
The cmdlet verifies previous success state AND resets it (side effect). Name of the cmdlet suggests that it provides only some verification. It can be misleading since it does not follow single-responsibility principle. Renaming like `CheckErrorOrResetState` would be clearer but a little bit ackward. IMHO this function and the main control flow should be redesigned to a simple strategy pattern (sequence of bool-returning cmdlets might work as well).

Reply:
Thanks for suggestion. Unfortunately I can't use bool returns, as the functions are suing Write-Output already which writes to STDOUT and also to RETURN of the function (Powershell "builtin" feature/quirks).
However renaming is a good idea. How about "Test-FailureOrReset"? **Test** is a verb suggested in powershell to check for something (e.g. Test-Path instead of FileExists)

Reply:
"Test-FailureOrReset" sounds fine.